services/sonar/sonar-documented-api-density.service.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const SonarBase = require('./sonar-base')
+const {
+  patternBase,
+  queryParamSchema,
+  getLabel,
+  goodMetricColorScale,
+} = require('./sonar-helpers')
+
+const metric = 'public_documented_api_density'
+
+module.exports = class SonarDocumentedApiDensity extends SonarBase {
```

Comment:
You haven't defined any examples on this class so it won't appear on the index page

Reply:
Understood, and that was done intentionally based on what was included/excluded from the Sonar example listings originally. 

I meant to pose a question here though on examples, as I'm not sure what the best approach is. Should we have an example for every single one? Should we maintain existing behavior of only listing a handful of examples for the more core/well-known metrics?

Reply:
Yeah I'm not really sure. For the `SonarGeneric` badge, I wouldn't want to explicitly list an individual example for every metric in the index, or show the URL in the index with

`:metric(new_violations|new_blocker_violations|new_critical_violations|new_major_violations|new_minor_violations.....)` or whatever in the URL pattern.

I think for that one we should probably list it with just `:metric` in the pattern and then explain in the documentation/help text how to find the valid values for `:metric` in the Sonar docs. Does that sound reasonable?

I'm less clear about `SonarDocumentedApiDensity` / `SonarFortifyRating`. If we're supplying a custom badge for it then I think we should probably list an example.

Reply:
Will do

Reply:
Example added